agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/README.md

```diff
@@ -0,0 +1,43 @@
+# Create Agents with code interpretation capabilities
+
+In this folder, we provide an example of an analytics agent using Agents for Amazon Bedrock new capabilities for [code interpretation](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-code-interpretation.html).
+
+The code interpreter is a sandboxed runtime environment in which the agent can run code. 
+
+In this example we will create a test agent with the following architecture:
+
+![Agent architecture](images/architecture.png)
+
+
+Code interpretation is made available to the agent via a pre-defined action group. Once the agent is created, you can enable the code interpretation capabilities via the `CreateAgentActionGroup` request using the `parentActionGroupSignature` parameter and setting it to `AMAZON.CodeInterpreter`.
+
+
+The code below shows how to configure the code interpretation capabilities when creating your agent's action group using the `create_agent_action_group` function from [boto3 SDK](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-agent/client/create_agent_action_group.html). You should set the  `parentActionGroupSignature` to `AMAZON.CodeInterpreter` and the `actionGroupState` status to `ENABLED`. Note that you must leave the description, apiSchema, and actionGroupExecutor fields blank for this action group.
+
+```python
+    response = client.create_agent_action_group(
+    actionGroupName='CodeInterpreterAction',
+    actionGroupState='ENABLED',
+    agentId='<YOUR_AGENT_ID>',
+    agentVersion='<YOUR_AGENT_VERSION>',
+    parentActionGroupSignature='AMAZON.CodeInterpreter'
+)
+```
+
+When invoking your agent, should supply parameter including the query text, agent id, agent alias, and a session id. Other parameters allow enabling tracing, to see the details of the model's return communication stream, ending the session, optionally storing conversational memory, and maintaining a state for this session.
```

Comment:
there is no mention on how to use code interpretation for inference here. This is for memory, needs to be updated for code interpretation

Reply:
The code interpretation doesn't actually need any parameters, if it's enabled via the special action group, the agent just decides if it wants to use it. I could add a note about that...